packages/ra-ui-materialui/src/input/ArrayInput/ArrayClearButton.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import { ButtonProps, IconButtonWithTooltip } from '../../button';
+import ClearIcon from '@mui/icons-material/HighlightOff';
+import { useSimpleFormIterator } from './useSimpleFormIterator';
+
+export const ArrayClearButton = (props: ButtonProps) => {
+    const { clear } = useSimpleFormIterator();
```

Comment:
I don't understand: should developers use the callback from the hook, or the `onClick` injected by SimpleFormIterator?

Reply:
You are right. 

developer should use the callback from the `onClick` injection 

commit 6c22695431f8cde914af7f48e4e4e8774b8c377d

Reply:
We shouldn't inject anything, better rely on the context

Reply:
`<ClearArrayButton />` is not doesn't have any injection